core/src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceService.scala

```diff
@@ -465,286 +467,116 @@ class WorkspaceService(
+      case WorkspaceType.RawlsWorkspace => ()
     }
-    workspaceChildren <- samDAO
-      .listResourceChildren(SamResourceTypeNames.workspace, workspace.workspaceId, ctx)
-      .map(
-        // a workspace may have a single child, if that child is the google project: this is deleted as part of the normal process
-        _.filter(c =>
-          c.resourceTypeName != SamResourceTypeNames.googleProject.value || workspace.googleProjectId.value != c.resourceId
-        )
-      )
-    googleProjectChildren <-
-      samDAO.listResourceChildren(SamResourceTypeNames.googleProject, workspace.googleProjectId.value, ctx)
-    blockingChildren = workspaceChildren.toList ::: googleProjectChildren.toList
-  } yield
-    if (blockingChildren.nonEmpty) {
-      val reports =
-        blockingChildren.map(r => ErrorReport(s"Blocking resource: ${r.resourceTypeName} resource ${r.resourceId}"))
-      throw RawlsExceptionWithErrorReport(
-        ErrorReport(StatusCodes.BadRequest, "Workspace deletion blocked by child resources", reports)
-      )
-    }
-
-  private def deleteWorkspaceInternal(workspaceContext: Workspace,
-                                      maybeMcWorkspace: Option[WorkspaceDescription],
-                                      parentContext: RawlsRequestContext
-  ): Future[WorkspaceDeletionResult] = {
-    if (isAzureMcWorkspace(maybeMcWorkspace)) {
-      return Future.failed(
-        new RawlsExceptionWithErrorReport(ErrorReport(StatusCodes.InternalServerError, "MC workspaces not supported"))
-      )
+    _ <- requesterPaysSetupService.deleteAllRecordsForWorkspace(workspace)
+    workflowsToAbort <- traceFutureWithParent("gatherWorkflowsToAbortAndSetStatusToAborted", ctx)(_ =>
+      submissionsRepository.getActiveWorkflowsAndSetStatusToAborted(workspace)
+    )
+    // Attempt to abort any running workflows so they don't write any more to the bucket.
+    // Notice that we're kicking off Futures to do the aborts concurrently, but we never collect their results!
+    // This is because there's nothing we can do if Cromwell fails, so we might as well move on and let the
+    // ExecutionContext run the futures whenever
+    aborts = traceFutureWithParent("abortRunningWorkflows", ctx)(_ =>
+      Future.traverse(workflowsToAbort)(wf => executionServiceCluster.abort(wf, ctx.userInfo))
+    )
+    _ <- traceFutureWithParent("deleteFastPassGrantsTransaction", ctx)(childContext =>
+      fastPassServiceConstructor(childContext).removeFastPassGrantsForWorkspace(workspace)
+    )
+    // notify leonardo so it can cleanup any dangling sam resources and other non-cloud state
+    _ <- traceFutureWithParent("notifyLeonardo", ctx)(_ =>
+      leonardoService.cleanupResources(workspace.googleProjectId, workspace.workspaceIdAsUUID, ctx)
+    )
+    // Delete Google Project
+    _ <- traceFutureWithParent("deleteGoogleProject", ctx)(_ => deleteGoogleProject(workspace.googleProjectId))
```

Comment:
There are technically still v1 workspaces in Rawls and this would bork all other v1 workspaces in a billing project by deleting the billing project's google project. Painful to have to leave it in given our stance on v1 workspaces, but as long as they're still around we probably need to keep the check

Reply:
If it was a v1 workspace, we'd never have gotten this far. 
The first operation is `getWorkspaceContextAndPermissions`, which only returns v2 workspaces. So calling this endpoint on a v1 workspace would already return a `NoSuchWorkspace` exception.

Note: IIRC, v1 workspaces are not returned or treated as valid targets for _any_ user-facing operation at this point.

Reply:
Ah, `getWorkspaceContextAndPermissions` was supposed to return v1 workspaces in contrast to `getV2WorkspaceContextAndPermissions`, but I see that changed a few months ago when we moved `MultiCloudWorkspaceService` over to the `WorkspaceRepository`. The idea was to allow users to _delete_ v1 workspaces, but not do anything else with them. Since the decision's been made for us, this is fine!